question_answering/question_answering.py

```diff
@@ -50,158 +405,284 @@ def _get_prompt_template(
-    context.logger.info(f"Loading model '{model}' and tokenizer into a pipeline")
-    pipeline = transformers.pipeline(
+    # Initialize a generation pipline and return:
+    pipe = transformers.pipeline(
         task="text-generation",
         model=model,
         tokenizer=tokenizer,
-        trust_remote_code=True,
-        model_kwargs=model_kwargs,
+        batch_size=batch_size,
     )
-    context.logger.info("Model loaded, pipeline created")
+    pipe.tokenizer.pad_token_id = model.config.eos_token_id
+    return pipe
 
-    # Go over the audio files and infer through the model:
-    if pathlib.Path(input_path).is_file():
-        input_path = [input_path]
-    else:
-        text_files_directory = pathlib.Path(input_path).absolute()
-        input_path = list(text_files_directory.rglob("*.*"))
-    for i, text_file in enumerate(
-        tqdm(
-            input_path,
-            desc="Generating answers",
-            unit="file",
-        )
-    ):
-        try:
-            # Read the text:
-            with open(text_file, "r") as fp:
-                text = fp.read()
-            # For each text, try multiple times as the llm might experience zero answers:
-            tryout = 0
-            for tryout in range(model_answering_tryouts):
-                # Infer through the llm:
-                sequences = pipeline(
-                    prompt_template.format(text),
-                    generation_config=generation_config,
-                    num_return_sequences=1,
-                    eos_token_id=tokenizer.eos_token_id,
+
+def _read_file_batch(
+    file_batch: List[pathlib.Path],
+    prompt_template: str,
+) -> List[str]:
+    batch = []
+    # Go over all files and read in usable format
+    for file in file_batch:
+        with open(file, "r", encoding="utf-8") as fp:
+            batch.append(prompt_template.format(fp.read()))
+    return batch
+
+
+def _list_me(list_to_check: list, name: str, length: int):
+    # Check if is list, turn to list if not
+    list_to_check = (
+        list_to_check if isinstance(list_to_check, list) else [list_to_check]
+    )
+    list_len = len(list_to_check)
+    # If not a list, or is a list of len 1 we duplicate for correct length
+    # If list in wrong length throw an error
+    if list_len != length:
+        if list_len == 1:
+            return list_to_check * length
+        else:
+            raise ValueError(
+                f"The argument value of '{name}' is not equal to the length of the given questions - {length}"
+            )
+    return list_to_check
+
+
+class QuestionHandler:
+    """
+    A class for handling questions answering for a given question type.
+    This class is used as a base class for all question types, and for default question type (regular question
+    answering without any special handling).
+    """
+
+    class ConfigKeys:
+        pass
+
+    def __init__(self, qustion_config: dict = None):
```

Comment:
The configuration can be passed as keyword arguments. This way it will also validate itself when nott given proper options.

```suggestion
    def __init__(self, **kwargs):
```